tobac/merge_split.py

```diff
@@ -0,0 +1,477 @@
+                temp = list(np.unique(temp))
+                arr = np.append(arr, np.unique(temp))
+
+                track_id[np.nanmax(np.unique(temp))] = list(np.unique(temp))
+
+    storm_id = [0]  # default because we don't track larger storm systems *yet*
+    print("found storm id")
+
+    track_parent_storm_id = np.repeat(
+        0, len(track_id)
+    )  # This will always be zero when we don't track larger storm systems *yet*
+    print("found track parent storm ids")
+
+    track_ids = np.array(list(track_id.keys()))
+    print("found track ids")
+
+    cell_id = list(
+        np.unique(
+            TRACK.cell.values.astype(float)[~np.isnan(TRACK.cell.values.astype(float))]
+        )
+    )
+    print("found cell ids")
+
+    cell_parent_track_id = []
+
+    for i, id in enumerate(track_id):
+
+        if len(track_id[int(id)]) == 1:
+            cell_parent_track_id.append(int(id))
+
+        else:
+            cell_parent_track_id.append(np.repeat(int(id), len(track_id[int(id)])))
+
+    cell_parent_track_id = list(flatten(cell_parent_track_id))
+    print("found cell parent track ids")
+
+    feature_parent_cell_id = list(TRACK.cell.values.astype(float))
+
+    print("found feature parent cell ids")
+
+    # This version includes all the feature regardless of if they are used in cells or not.
+    feature_id = list(TRACK.feature.values.astype(int))
+    print("found feature ids")
+
+    feature_parent_storm_id = np.repeat(0, len(feature_id))  # we don't do storms atm
+    print("found feature parent storm ids")
+
+    feature_parent_track_id = []
+    feature_parent_track_id = np.zeros(len(feature_id))
+    for i, id in enumerate(feature_id):
+        cellid = feature_parent_cell_id[i]
+        if np.isnan(cellid):
+            feature_parent_track_id[i] = -1
+        else:
+            j = np.where(cell_id == cellid)
+            j = np.squeeze(j)
+            trackid = cell_parent_track_id[j]
+            feature_parent_track_id[i] = trackid
+
+    print("found feature parent track ids")
+
+    storm_child_track_count = [len(track_id)]
+    print("found storm child track count")
+
+    track_child_cell_count = []
+    for i, id in enumerate(track_id):
+        track_child_cell_count.append(len(track_id[int(id)]))
+    print("found track child cell count")
+
+    cell_child_feature_count = []
+    for i, id in enumerate(cell_id):
+        cell_child_feature_count.append(len(track_groups[id].feature.values))
+    print("found cell child feature count")
+
+    storm_child_cell_count = [len(cell_id)]
+    storm_child_feature_count = [len(feature_id)]
+
+    storm_dim = "nstorms"
+    track_dim = "ntracks"
+    cell_dim = "ncells"
+    feature_dim = "nfeatures"
+
+    d = xr.Dataset(
+        {
+            "storm_id": (storm_dim, storm_id),
+            "track_id": (track_dim, track_ids),
+            "track_parent_storm_id": (track_dim, track_parent_storm_id),
+            "cell_id": (cell_dim, cell_id),
```

Comment:
I think we should keep the names as just `cell`, `track`, etc. @JuliaKukulies what do you think?

Reply:
Yes, agree with this